sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,341 @@
+from __future__ import division
+
+__author__ = "Henry Lin <hlin117@gmail.com>"
+
+import numpy as np
+import scipy.sparse as sp
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from sklearn.utils.sparsefuncs import csr_csc_min_axis0, csr_csc_max_axis0
+from _discretization import binsearch
+from six.moves import xrange
+
+__all__ = [
+    "Discretizer"
+]
+
+
+class Discretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int (default=2)
+        The number of bins to produce. The intervals for the bins are
+        determined by the minimum and maximum of the input data.
+        Raises ValueError if n_bins < 2.
+
+    categorical_features : int, array-like, or None (default=None)
+        Specifies the indices of categorical columns which are not to
+        be discretized. If None, assumes that all columns are continuous
+        features.
+
+    Attributes
+    ----------
+    min_ : float
+        The minimum value of the input data.
```

Comment:
Yes, for each non-discrete (non-categorical / continuous) column. I should probably be more explicit about this.